src/hid.c

```diff
@@ -148,6 +148,36 @@ fido_dev_info_ptr(const fido_dev_info_t *devlist, size_t i)
 	return (&devlist[i]);
 }
 
+int
+fido_dev_info_set(fido_dev_info_t *devlist, size_t i,
+    const char *path, const char *manufacturer, const char *product,
+    const fido_dev_io_t *io, const fido_dev_transport_t *transport)
+{
+	char *path_copy = NULL, *manu_copy = NULL, *prod_copy = NULL;
+	int r = FIDO_ERR_INTERNAL;
+
+	if ((path && (path_copy = strdup(path)) == NULL) ||
```

Comment:
`fido_dev_open_with_info()` does not accept a `NULL` path, perhaps we should return `FIDO_ERR_INVALID_ARGUMENT` for that case here?

For devices without `manufacturer` and `product` strings, we use an empty string rather than `NULL` (see #414). Perhaps it'd make sense to align with that behavior here as well.